src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -0,0 +1,97 @@
+import assert from "assert";
+import * as uma from "@uma/sdk";
+import { Provider } from "@ethersproject/providers";
+import { BigNumber } from "ethers";
+import { BigNumberish, calculateGasFees, toBNWei } from "../utils";
+const Coingecko = uma.Coingecko;
+const { percent, fixedPointAdjustment } = uma.across.utils;
+const { connect: erc20Connect } = uma.clients.erc20;
+
+export interface QueryInterface {
+  getGas: (tokenAddress?: string) => Promise<BigNumberish>;
+  getGasPrice: () => Promise<BigNumberish>;
+  getTokenPrice: (tokenAddress: string) => Promise<number | string>;
+  getTokenDecimals: (tokenAddress: string) => Promise<number>;
```

Comment:
these are the specific calls needed to calculate the fees, but might be different per chain